catalog/dags/data_refresh/distributed_reindex.py

```diff
@@ -0,0 +1,425 @@
+    )
+
+
+@task(
+    # Run locally when create instance tasks are skipped
+    trigger_rule=TriggerRule.NONE_FAILED
+)
+def get_instance_ip_address(
+    environment: str, instance_id: str, aws_conn_id: str = AWS_CONN_ID
+):
+    if environment != PRODUCTION:
+        return "catalog_indexer_worker"
+
+    ec2_hook = EC2Hook(aws_conn_id=aws_conn_id, api_type="client_type")
+    reservations = ec2_hook.describe_instances(instance_ids=[instance_id]).get(
+        "Reservations"
+    )
+
+    # `Reservations` is a list of dicts, grouping instances by the launch request that
+    # started them. Because we are querying a single InstanceId we expect to have one
+    # Reservation.
+    if not (len(reservations) == 1 and len(reservations.get("Instances", {})) == 1):
+        raise Exception("Unable to describe worker instance.")
+
+    return reservations[0].get("Instances", {})[0].get("PrivateIpAddress")
+
+
+@task(
+    # Instance creation tasks are skipped locally, but we still want this task to run.
+    trigger_rule=TriggerRule.NONE_FAILED
+)
+def create_connection(
+    environment: str,
+    target_environment: str,
+    instance_id: str,
+    server: str,
+):
+    """
+    Create an Airflow Connection for the given indexer worker and persist it. It will
+    later be dropped in a cleanup task.
+    """
+    worker_conn_id = f"indexer_worker_{instance_id or 'localhost'}"
+
+    # Create the Connection
+    logger.info(f"Creating connection with id {worker_conn_id}")
+    worker_conn = Connection(conn_id=worker_conn_id, uri=f"http://{server}:8003/")
+    session = settings.Session()
+    session.add(worker_conn)
+    session.commit()
+
+    return worker_conn_id
+
+
+@task
+def terminate_indexer_worker(
+    environment: str,
+    instance_id: str,
+    aws_conn_id: str = AWS_CONN_ID,
+):
+    """Terminate an individual indexer worker."""
+    if environment != PRODUCTION:
+        raise AirflowSkipException(
+            "Skipping instance termination in local environment."
+        )
+
+    ec2_hook = EC2Hook(aws_conn_id=aws_conn_id, api_type="client_type")
+    return ec2_hook.conn.terminate_instances(instance_ids=[instance_id])
+
+
+@task(trigger_rule=TriggerRule.ALL_DONE)
+def drop_connection(worker_conn: str):
+    """Drop the Connection to the now terminated instance."""
+    conn = Connection.get_connection_from_secrets(worker_conn)
+
+    session = settings.Session()
+    session.delete(conn)
+    session.commit()
+
+
+@task_group(group_id="reindex")
+def reindex(
+    data_refresh_config: DataRefreshConfig,
+    target_index: str,
+    start_id: int,
+    end_id: int,
+    environment: str,
+    target_environment: Environment,
+    aws_conn_id: str = AWS_CONN_ID,
+):
+    """
+    Trigger a reindexing task on a remote indexer worker and wait for it to complete. Once done,
+    terminate the indexer worker instance.
+    """
+
+    launch_template_version = get_launch_template_version(
```

Comment:
Because launch template versions are full resources, it may be helpful to disambiguate the names for this and the task the way AWS itself does, by specifying that it's a version _number_, rather than a full launch template version document.